pkg/customNoRouteHandler/customNoRouteHandler.go

```diff
@@ -0,0 +1,78 @@
+package customNoRouteHandler
+
+import (
+	"io"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/luraproject/lura/logging"
+)
+
+type NoRouteHandler struct {
+	httpClient    *http.Client
+	defaultURL    string
+	defaultScheme string
+	logger        logging.Logger
+}
+
+func NewNoRouteHandler(defaultURL, defaultScheme string, logger logging.Logger) *NoRouteHandler {
+	h := &NoRouteHandler{
+		defaultURL:    defaultURL,
+		defaultScheme: defaultScheme,
+		logger:        logger,
+	}
+
+	h.InitialiseHttpClient()
+
+	return h
+}
+
+func (h *NoRouteHandler) InitialiseHttpClient() {
+	transport := &http.Transport{
+		MaxIdleConns:        100,
```

Comment:
these should be tuned based on load testing

Reply:
Sure, thanks. maxIdleConns is currently the default value and increased maxIdleConnsPerHost from default value of 2 to 100. Will be fine-tuning these values after gamma deployment and load testing.